### ISSUE TITLE
Custom request interception does not work with Pa11y headers overrides

### ISSUE THREAD

**Issue by thibaudcolas** [pa11y/pa11y#454]
## Expected behaviour

When providing custom `browser` / `page` instances to Pa11y, it should be possible to do custom request interception to make accessibility tests that rely on specific network conditions (with Puppeteer’s `page.setRequestInterception(true)`, and the `request` event on pages) or data from third-party tools.

## Actual behaviour

This works already… unless you also happen to be using the `headers` overrides option to set cookies, or other HTTP headers. Then, Pa11y's request interception conflicts with what's set outside of Pa11y.

The actual error is: `Error: Request is already handled!` on the `request.continue(overrides)` call.

## Steps to reproduce

1. Use Pa11y with `headers` overrides (e.g. cookies)
2. Set up custom `browser` and `page` instances forward to Pa11y via the options.
3. Add custom request interception, for example aborting requests to a specific AJAX API call.

## Environment:

```
Pa11y:      5.1.0
Node.js:    10.15.2
npm:        6.4.1
OS:         18.2.0 (darwin)
```

---

My solution to this was to stop using the `headers` option from Pa11y, and re-implement this logic in my custom network interception code. Here is what it looks like roughly:

```js
const headers = {
  Cookie: `sessionid=${sharedCookie.value};`,
};

await page.setRequestInterception(true);

let interceptionHandled = false;
page.on("request", (request) => {
  // This is normally implemented by pa11y. We do this ourselves because it would be incompatible with any other request override.
  const overrides = {};
  if (!interceptionHandled) {
    overrides.headers = {};
    for (const [key, value] of Object.entries(headers)) {
      overrides.headers[key.toLowerCase()] = value;
    }

    interceptionHandled = true;
  }

  if (scenario_is_relevant) {
    /* Your custom request interception logic */
    return;
  }

  request.continue(overrides);
});

const result = await pa11y(`${ADMIN_ROOT}${scenario.path}`, pa11yOptions);
```

**Comment by joeyciechanowicz** [comment]
Beautifully written bug report! Thanks.

A fourth option is to not perform request interception if a `page` is passed in to `pa11y`. On the basis that a user has performed some custom setup if they're going to the trouble of passing in a `page`.
